source/client-side-encryption/tests/README.rst

```diff
@@ -2693,302 +2694,343 @@ Create a ClientEncryption object named ``clientEncryption`` with these options:
    }
 
-Ensure 201 matches the type of the encrypted field. The error should be raised because 201 is greater than the maximum value in ``RangeOpts``.
+Assert that an error was raised because 201 is greater than the maximum value in ``RangeOpts``.
 
-Assert that an error was raised.
 
-Case 7: encrypting a document of a different type errors 
-````````````````````````````````````````````````````````
+Case 7: encrypting a value of a different type errors
+`````````````````````````````````````````````````````
 This test case should be skipped if the encrypted field is ``encryptedDoubleNoPrecision`` or ``encryptedDecimalNoPrecision``.
 
-For all the tests below use these ``EncryptOpts``:
+Use ``clientEncryption.encrypt()`` to encrypt the value 6 with a type that does not match that of the encrypted field.
+
+If the encrypted field is ``encryptedInt`` use a BSON double type. Otherwise, use a BSON int32 type.
+
+Encrypt using the following ``EncryptOpts``:
 
 .. code:: typescript
 
    class EncryptOpts {
-      keyId : <key1ID>
+      keyId : <key1ID>,
       algorithm: "RangePreview",
-      contentionFactor: 0
+      contentionFactor: 0,
+      rangeOpts: <RangeOpts for Type>,
    }
 
-If the encrypted field is ``encryptedInt`` encrypt ``{ "encryptedInt": { "$numberDouble": "6" } }``.
-Otherwise, encrypt ``{ "encrypted<Type>": { "$numberInt": "6" }``.
-Assert an error was raised.
+Ensure that ``RangeOpts`` corresponds to the type of the encrypted field (i.e. expected type) and not that of the value being passed to ``clientEncryption.encrypt()``.
+
+Assert that an error was raised.
+
 
+Case 8: setting precision errors if the type is not a floating point
+````````````````````````````````````````````````````````````````````
+This test case should be skipped if the encrypted field is ``encryptedDoublePrecision``, ``encryptedDoubleNoPrecision``, ``encryptedDecimalPrecision``, or ``encryptedDecimalNoPrecision``.
 
-Case 8: setting precision errors if the type is not a double
-````````````````````````````````````````````````````````````
-This test case should be skipped if the encrypted field is ``encryptedDoublePrecision`` or ``encryptedDoubleNoPrecision`` or ``encryptedDecimalPrecision`` or ``encryptedDecimalNoPrecision``.
+Use ``clientEncryption.encrypt()`` to encrypt the value 6. Ensure the type matches that of the encrypted field.
 
-Use ``clientEncryption.encrypt()`` to try to encrypt the value 6 with these ``EncryptOpts`` and these ``RangeOpts``:
+Add ``{ precision: 2 }`` to the encrypted field's ``RangeOpts`` (see: `Test Setup: RangeOpts`_).
+
+Encrypt using the following ``EncryptOpts``:
 
 .. code:: typescript
 
    class EncryptOpts {
-      keyId : <key1ID>
+      keyId : <key1ID>,
       algorithm: "RangePreview",
-      contentionFactor: 0
-   }
-   
-   class RangeOpts {
-      min: 0,
-      max: 200,
-      sparsity: 1,
-      precision: 2,
+      contentionFactor: 0,
+      rangeOpts: <RangeOpts for Type with precision added>,
```

Comment:
The previous text didn't mention anything about using RangeOpts for the specific type. I think this makes it more explicit and ensures the min/max values with have the correct type.

Reply:
Following up: I realize now that this actually changes the test behavior.

I originally thought that the use of fixed RangeOpts options was a mistake. It see now that this is relying on libmongocrypt validating the `precision` field _before_ checking types based on `min` and `max`. That seems pretty fragile, but I'm OK to revert this in order to avoid any downstream changes.

Please let me know what you think.

Reply:
Ok, I'll keep this in place. And it sounds like you don't care to notify drivers of this clarification. I suppose if anything breaks down the line (change in libmongocrypt's validation order), they can come back to this test and easily correct the tests.